pandas/tseries/tools.py

```diff
@@ -283,10 +303,25 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     NaT
 
     """
-    return _to_datetime(arg, errors=errors, dayfirst=dayfirst,
-                        yearfirst=yearfirst,
-                        utc=utc, box=box, format=format, exact=exact,
-                        unit=unit, infer_datetime_format=infer_datetime_format)
+    # variable to set offset as per origin parameter
+    offset = None
+
+    if origin == 'julian':
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        arg = arg - tslib.Timestamp(0).to_julian_date()
```

Comment:
I c. ok

then we should catch these out-of-ranges and either return actual datetimes or raise a nicer error message (I think the later is simply better)

```
In [1]: pd.to_datetime(['1000-01-01'])
OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 1000-01-01 00:00:00
```

Furthermore should be able to handle this situation (and need tests).
```
In [2]: pd.to_datetime(['1000-01-01'],errors='coerce')
Out[2]: DatetimeIndex(['NaT'], dtype='datetime64[ns]', freq=None)
```